gap/oper.gi

```diff
@@ -249,8 +259,19 @@ end);
 InstallMethod(DigraphRemoveEdge,
 "for a immutable digraph and two positive integers",
 [IsImmutableDigraph, IsPosInt, IsPosInt],
-{D, src, ran}
--> MakeImmutable(DigraphRemoveEdge(DigraphMutableCopy(D), src, ran)));
+function(D, src, ran)
+  local pos, DMutable;
+  DMutable := DigraphMutableCopy(D);
+  DIGRAPHS_CheckDigraphRemoveEdge(DMutable, src, ran);
```

Comment:
Here we create a copy and then check it is valid.  Why not check the original graph *before* copying, so that extra work is not done?